src/core/server/jest.integration.config.js

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module.exports = {
+  preset: '@kbn/test/jest_integration_node',
```

Comment:
This is the root change of the PR.
By changing to a _node-like_ preset for _server-side_ tests, we aim at solving the `Cross origin error` encountered [here](https://github.com/elastic/kibana/issues/123666#issuecomment-1098932363).

By performing this change, a bunch of errors emerged, which make other tests and type checks fail. These are tackled within this PR.